### PR TITLE
fix: Windows and VMSS version validation for azs

### PR DESCRIPTION
**Pull request by haofan-ms**
**Reason for Change**:

The windows validation failed on AzureStack as the validation was using the version list for Azure. This was caused by some hard-coded changes before, thus update this, as well as the VMSS validation.

**Issue Fixed**:

**Credit Where Due**:

Does this change contain code from or inspired by another project?

- [ ] No
- [ ] Yes

If "Yes," did you notify that project's maintainers and provide attribution?

- [ ] No
- [ ] Yes

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: pkg/api/vlabs/validate.go [hunk 1]
```diff
@@ -1172,8 +1172,8 @@ func (a *Properties) validateWindowsProfile(isUpdate bool) error {
 			o.OrchestratorRelease,
 			o.OrchestratorVersion,
 			isUpdate,
-			true,
-			false)
+			a.HasWindows(),
```
[the remark this answers is not in the document]
Comment (haofan-ms) [reply pkg/api/vlabs/validate.go]: Updated

File: pkg/api/vlabs/validate.go [hunk 2]
```diff
@@ -194,8 +194,8 @@ func (a *Properties) ValidateOrchestratorProfile(isUpdate bool) error {
 				o.OrchestratorRelease,
 				o.OrchestratorVersion,
 				isUpdate,
-				false,
-				false)
+				a.HasWindows(),
```
[the remark this answers is not in the document]
Comment (haofan-ms) [reply pkg/api/vlabs/validate.go]: Thanks, reverted.

File: pkg/api/vlabs/validate.go [hunk 3]
```diff
@@ -1103,15 +1103,15 @@ func (a *AgentPoolProfile) validateCustomNodeLabels(orchestratorType string) err
...
+func validateVMSS(o *OrchestratorProfile, isUpdate bool, storageProfile string, hasWindows bool, isAzureStackCloud bool) error {
 	if o.OrchestratorType == Kubernetes {
 		version := common.RationalizeReleaseAndVersion(
 			o.OrchestratorType,
 			o.OrchestratorRelease,
 			o.OrchestratorVersion,
 			isUpdate,
-			false,
-			false)
+			hasWindows,
```
[the remark this answers is not in the document]
Comment (haofan-ms) [reply pkg/api/vlabs/validate.go:1113]: For Azure, the difference between linux and windows support are only applying to the old k8s versions. 
@jackfrancis can you help confirm this change is okay?

[the remark this answers is not in the document]
Comment (jackfrancis) [reply pkg/api/vlabs/validate.go:1113]: I think this change makes sense. Without it (the way it is now) we are subject to edge cases where we allow a version not supported in Windows (rare) or Azure Stack (less rare) in this flow.

Although, I'm not confident this particular VMSS validation flow is the last line of defense for validating versions.

tl;dr this change is safe.

**Review by jadarsie — DISMISSED** [review]

**Review by jackfrancis — APPROVED** [review]
/lgtm
